command/init.go

```diff
@@ -526,6 +526,21 @@ func (c *InitCommand) getProviders(config *configs.Config, state *states.State,
 			}
 
 		},
+		QueryPackagesWarning: func(provider addrs.Provider, warnings []string) {
+			displayWarnings := make([]string, len(warnings))
+			for i, warning := range warnings {
+				displayWarnings[i] = fmt.Sprintf("- %s", warning)
+			}
+
+			diags = diags.Append(tfdiags.Sourceless(
+				tfdiags.Warning,
+				"",
```

Comment:
The use of an empty string for the warning here makes me a little uneasy. I can't find any other instances where we have a blank summary for a diag, and I wonder about some different future UI presentation of diags not coping well with this.

Some possible ideas to riff on in case you decide to add text here:

- "Warning: Provider installation succeeded with warnings"
- "Warning: Provider installation includes alerts"
- "Warning: Additional provider information from registry"

There is also the option of adding one diag per warning and using the text from the registry as the summary.

Reply:
I like your first suggestion best, except that we won't actually know if installation succeeded. I tried a few different options and they all looked super redundant. We also don't know how many warnings we'll get, which is why I didn't do a full diag for each. The output looked better without a summary, imo, but I'll add something 🤷